parlai/core/torch_agent.py

```diff
@@ -0,0 +1,260 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+from parlai.core.agents import Agent
+from parlai.core.dict import DictionaryAgent
+
+try:
+    import torch
+except Exception as e:
+    raise ModuleNotFoundError('Need to install Pytorch: go to pytorch.org')
+
+from collections import deque, namedtuple
+import random
+
+Batch = namedtuple("Batch", ["text_vecs", "label_vecs", "labels",
+                             "valid_inds"])
```

Comment:
```
Batch = namedtuple("Batch", [
    "text_vec",  # bsz x seqlen tensor containing the parsed text data
    "label_vec", # bsz x seqlen tensor containing the parsed label (one per batch row)
    "labels", # list of length bsz containing the selected label for each batch row (some datasets have multiple labels per input example)
    "valid_indices",  # list of length bsz containing the original indices of each example in the batch. we use these to map predictions back to their proper row, since e.g. we may sort examples by their length or some examples may be invalid.
])
```